src/screens/Signup/index.test.tsx

```diff
@@ -5,6 +5,10 @@ import { setupServer } from 'msw/node';
 
 import Signup from '.';
 
+const mockHistoryPush = jest.fn();
+
+jest.mock('react-router-dom', () => ({ useHistory: () => ({ push: mockHistoryPush }) }));
```

Comment:
Don't know if this is exactly the case but I believe that it's worth noting: 
Beware with mocking an entire library like this because you are "overriding" it, in this test 'react-router-dom' will return an object with a single method called useHistory that returns an object. If some of the elements in your test require another method from 'react-router-dom' you will get an error compiling the test because those methods won't be found.

For more info check out `jest.requireActual`